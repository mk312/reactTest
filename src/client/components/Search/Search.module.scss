.wrapper {
  position: relative;
  min-height: 150px;
  padding: 100px 75px 75px;

  &:after {
    content: '';
    position: absolute;
    z-index: -1;
    left:0;
    right: 0;
    top: 0;
    bottom: 0;
    background-image: url('/search_bg.jpg');
    background-size: cover;
    filter: brightness(0.35) contrast(0.65) blur(1px);
  }
}

.row {
  display: flex;
  margin-bottom: 40px;
  align-items: baseline;
}

.title {
  margin: 0 0 20px;
  color: #ffffff;
  font-size: 25px;
}

.input {
  width: calc(85% - 10px);
  margin-right: 10px;
}

.submit {
  min-width: 15%;
}

.input,
.submit {
  padding: 14px;
  border: 0;
  outline: none;
  font-size: 13px;
  color: #ffffff;
}

.filter {
  display: inline-block;
  margin-right: 14px;
}
