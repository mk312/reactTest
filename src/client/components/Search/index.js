import img from './search_bg.jpg';
import styles from './styles.scss';

import React, { PureComponent } from 'react';
import Toggle from '../basic/Toggle/';

export default class Search extends PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            searchValue: 'Quentin Tarantino',
            filterValue: 'TITLE',
        };
        this.handleSearchChange = (event) => {
            this.setState({searchValue: event.target.value});
        };
        this.handleFilterChange = (event) => {
            this.setState({filterValue: event.target.value});
        };
        this.handleSubmit = (event) => {
            event.preventDefault();
            this.props.searchMovies({
                searchValue: this.state.searchValue,
                filterValue: this.state.filterValue,
            });
        };
    }

    render() {
        return (
           <form className={styles.wrapper} onSubmit={this.handleSubmit}>
               <div className={'row'}>
                   <div className={styles.title}>FIND YOUR MOVIE</div>
                   <input className={styles.input} type="text" value={this.state.searchValue} onChange={this.handleSearchChange} />
                   <input className={styles.submit} type="submit" value="SEARCH"/>
               </div>

               <div className={'row'}>
                   <div className={styles.filter}>SEARCH BY {this.state.filterValue}</div>
                   <Toggle handleFilterChange={this.handleFilterChange}
                           filterValue={this.state.filterValue}
                           values={[{value: 'TITLE', id: 'filterValue1'}, {value: 'GENRE', id: 'filterValue2'}]} />
               </div>
           </form>
        );
    }
}