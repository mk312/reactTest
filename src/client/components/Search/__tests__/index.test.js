import React from 'react';
import {Simulate} from 'react-dom/test-utils';
import {render} from '@testing-library/react'

import Search from '../index';

let container = null;

describe('App', () => {
    beforeEach(() => {
        container = render(<Search />);
    });
    it('should toggle between Genre/Title when one of them is clicked', () => {
        const radio = [
                {id:'filterValue1', value: 'TITLE'},
                {id:'filterValue2', value: 'GENRE'},
            ]
        radio.forEach ((item) => {
            const toggleButton =document.getElementById(`${item.id}`);
            Simulate.change(toggleButton, { target: { value: item.value }});

            expect(toggleButton).toHaveProperty("checked");
            // re-check with snapshot
            expect(container).toMatchSnapshot();
        })
    });
});