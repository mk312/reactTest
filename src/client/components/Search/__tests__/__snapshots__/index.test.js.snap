// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should toggle between Genre/Title when one of them is clicked 1`] = `
<div>
  <form
    class="js-search-form wrapper"
  >
    <div
      class="row"
    >
      <div
        class="title"
      >
        FIND YOUR MOVIE
      </div>
      <input
        class="js-search-input input"
        type="text"
        value=""
      />
      <input
        class="submit"
        type="submit"
        value="SEARCH"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="filter"
      >
        SEARCH BY 
        TITLE
      </div>
      <input
        checked=""
        class="js-toggle-radio-button"
        id="filterValue1"
        name="contact"
        type="radio"
        value="TITLE"
      />
      <label
        for="filterValue1"
      >
        TITLE
      </label>
      <input
        class="js-toggle-radio-button"
        id="filterValue2"
        name="contact"
        type="radio"
        value="GENRE"
      />
      <label
        for="filterValue2"
      >
        GENRE
      </label>
    </div>
  </form>
</div>
`;

exports[`App should toggle between Genre/Title when one of them is clicked 2`] = `
<div>
  <form
    class="js-search-form wrapper"
  >
    <div
      class="row"
    >
      <div
        class="title"
      >
        FIND YOUR MOVIE
      </div>
      <input
        class="js-search-input input"
        type="text"
        value=""
      />
      <input
        class="submit"
        type="submit"
        value="SEARCH"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="filter"
      >
        SEARCH BY 
        GENRE
      </div>
      <input
        class="js-toggle-radio-button"
        id="filterValue1"
        name="contact"
        type="radio"
        value="TITLE"
      />
      <label
        for="filterValue1"
      >
        TITLE
      </label>
      <input
        checked=""
        class="js-toggle-radio-button"
        id="filterValue2"
        name="contact"
        type="radio"
        value="GENRE"
      />
      <label
        for="filterValue2"
      >
        GENRE
      </label>
    </div>
  </form>
</div>
`;
